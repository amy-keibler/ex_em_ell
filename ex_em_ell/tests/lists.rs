#[derive(Debug, PartialEq, ex_em_ell::FromXmlDocument, ex_em_ell::ToXmlDocument)]
struct Example {
    children: Vec<ExampleChild>,
}

#[derive(
    Debug, PartialEq, ex_em_ell::FromXmlElement, ex_em_ell::ToXmlElement, ex_em_ell::NamedXmlElement,
)]
#[ex_em_ell(name = "child")]
struct ExampleChild {
    field: String,
}

#[test]
fn test_example_xmls() {
    insta::glob!("data/lists/valid_*.xml", |path| {
        let file = std::fs::File::open(path).expect(&format!("Failed to read file: {path:?}"));
        let example: Example = ex_em_ell::from_reader(&file)
            .expect(&format!("Failed to parse the XML file: {path:?}"));

        let round_trip = ex_em_ell::to_string_pretty(&example).expect("Failed to output XML");
        insta::assert_snapshot!(round_trip);
    });
}
