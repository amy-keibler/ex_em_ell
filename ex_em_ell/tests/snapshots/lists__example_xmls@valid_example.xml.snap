---
source: ex_em_ell/tests/lists.rs
expression: round_trip
input_file: ex_em_ell/tests/data/lists/valid_example.xml
---
<?xml version="1.0" encoding="utf-8"?>
<example>
  <children>
    <child>
      <field>value</field>
    </child>
  </children>
</example>
